src/test/scala/uk/gov/nationalarchives/dp/client/PreservicaClientCacheTest.scala

```diff
@@ -0,0 +1,192 @@
+package uk.gov.nationalarchives.dp.client
+
+import cats.Applicative
+import cats.effect.kernel.Sync
+import org.mockito.ArgumentMatchers.{any, eq => mockEq}
+import org.mockito.{Mockito, MockitoSugar}
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers._
+
+import java.nio.file.Path
+import scala.concurrent.duration.{Duration, DurationInt}
+
+abstract class PreservicaClientCacheTest[F[_]: Sync: Applicative]()
+    extends AnyFlatSpec
+    with MockitoSugar
+    with BeforeAndAfterEach {
+  def defaultReadStringMock: Path => String = {
+    val readStringMock = mock[Path => String]
+    when(readStringMock.apply(any[Path])).thenReturn("cachedValue")
+    readStringMock
+  }
+  def defaultGetAttributeMock: (Path, String) => Array[Byte] = {
+    val getAttributeMock = mock[(Path, String) => Array[Byte]]
+    when(getAttributeMock.apply(any[Path], mockEq("user:ttl"))).thenReturn("10".getBytes)
+    when(getAttributeMock.apply(any[Path], mockEq("user:entry"))).thenReturn("10".getBytes)
+    getAttributeMock
+  }
+
+  def defaultSetAttributeMock: (Path, String, Array[Byte]) => Path = {
+    val setAttributeMock = mock[(Path, String, Array[Byte]) => Path]
+    when(setAttributeMock.apply(any[Path], mockEq("user:ttl"), any[Array[Byte]])).thenReturn(Path.of("/tmp"))
+    when(setAttributeMock.apply(any[Path], mockEq("user:entry"), any[Array[Byte]])).thenReturn(Path.of("/tmp"))
+    setAttributeMock
+  }
+
+  def defaultWriteMock: (Path, Array[Byte]) => Path = {
+    val writeMock = mock[(Path, Array[Byte]) => Path]
+    when(writeMock.apply(any[Path], any[Array[Byte]])).thenReturn(Path.of("/tmp"))
+    writeMock
+  }
+
+  def defaultDeleteMock: Path => Unit = {
+    val deleteMock = mock[Path => Unit]
+    when(deleteMock.apply(any[Path])).thenReturn(())
+  }
+
+  override def beforeEach(): Unit = {
+    Mockito.reset(
+      defaultWriteMock,
+      defaultSetAttributeMock,
+      defaultGetAttributeMock,
+      defaultReadStringMock,
+      defaultDeleteMock
+    )
+  }
+
+  val mockFiles: List[Path] = List(Path.of("path1"), Path.of("path2"))
+
+  case class TestClass(
```

Comment:
Should this class be called something more descriptive like `PreservicaClientCacheMock` or `MockPreservicaClientCache`?

Reply:
Can do, I've gone with `MockPreservicaClientCache`